### DISCUSSION TITLE
Separating latency report for read and write requests

### DISCUSSION THREAD

**Discussion by bigplanewangyue** [sarchlab/mgpusim#104] [Q&A]
How can I distinguish the task time for read and write operations in `averagetimetracer.go`?  And in `metrics.cvs` ,  `req_average_latency` seems to be the average of all the latencies for a specific location？

**Comment by JunjoFor** [comment]
If you print memory traces with `-trace-mem`  you can get the time each memory task started and ended so you can calculate the latencies you want.

**Comment by syifan** [comment]
The solution should be creating two different tracers with different trace filters. See here for the filter. https://github.com/sarchlab/mgpusim/blob/v3/samples/runner/report.go#L180. You can try to print the task and see what information is available in the filter. But I think the `What` field should be able to separate the read and write requests. 

I have also updated the related parts in the Akita book. https://www.notion.so/syifan/2-4-Hooking-and-Tracing-Ongoing-0b947051beb74868a2a74de81bafe40c?pvs=4#12919baeaf86804ea2c3c1da13af059d. I hope it is helpful.

**Reply by bigplanewangyue** [reply]
when i print the what and where field , i found they are null？

**Reply by syifan** [reply]
Are you printing the task in the `EndTask` function. The task passed to the `EndTask` only contains the task ID and end time. You may need to match it with the original task to identify the `where` field.

**Comment by bigplanewangyue** [comment]
thank you very much




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "sarchlab/mgpusim"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月24日(星期天) 晚上10:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [sarchlab/mgpusim] Separating latency report for read and write requests (Discussion #104)





 
Are you printing the task in the EndTask function. The task passed to the EndTask only contains the task ID and end time. You may need to match it with the original task to identify the where field.
